tracer/test/Datadog.Trace.Tests/Telemetry/TelemetryControllerTests.cs

```diff
@@ -87,6 +86,41 @@ public async Task TelemetryControllerDisposesOnTwoFatalErrorsFromTelemetry()
             transport.GetData().Count.Should().Be(previousDataCount.Count, "Should not send more data after disposal");
         }
 
+        [Fact]
+        public async Task TelemetrySendsHeartbeatsIndependentlyOfData()
+        {
+            var heartBeatInterval = TimeSpan.FromMilliseconds(100);
+            var transport = new TestTelemetryTransport(pushResult: TelemetryPushResult.Success);
+            var controller = new TelemetryController(
+                new ConfigurationTelemetryCollector(),
+                new DependencyTelemetryCollector(),
+                new IntegrationTelemetryCollector(),
+                transport,
+                flushInterval: TimeSpan.FromMinutes(1),
+                heartBeatInterval: heartBeatInterval);
+
+            controller.RecordTracerSettings(new ImmutableTracerSettings(new TracerSettings()), "DefaultServiceName", EmptyAasMetadata);
+            controller.Start();
+
+            var requiredHeartbeats = 10;
+            var deadline = DateTimeOffset.UtcNow.AddSeconds(heartBeatInterval.TotalSeconds * 100);
+            while (DateTimeOffset.UtcNow < deadline)
+            {
+                var heartBeatCount = transport.GetData().Count(x => x.RequestType == TelemetryRequestTypes.AppHeartbeat);
+                if (heartBeatCount > requiredHeartbeats)
```

Comment:
Should this be greater than or equal to here?
```suggestion
                if (heartBeatCount >= requiredHeartbeats)
```